index.js

```diff
@@ -155,27 +159,36 @@ module.exports.resizeBuffer = function(buffer, args, callback) {
 						image.embed();
 					} else if (args.w || args.h) {
 						image.resize(
-							Number(args.w) || null,
-							Number(args.h) || null
+							(Number(args.w) * zoom) || null,
+							(Number(args.h) * zoom) || null
 						);
 						if (!args.crop) {
 							image.max();
 						}
 					}
 
+					// return a default compression value based on a logarithmic scale
+					// defaultValue = 100, zoom = 2; = 65
+					// defaultValue = 80, zoom = 2; = 50
+					// defaultValue = 100, zoom = 1.5; = 86
+					// defaultValue = 80, zoom = 1.5; = 68
+					var applyZoomCompression = function( defaultValue, zoom ) {
+						return Math.min(Math.max(Math.round( defaultValue - ( (Math.log(zoom) / Math.log(defaultValue / zoom)) * (defaultValue * zoom) ) ), Math.round(defaultValue / zoom)), defaultValue);
```

Comment:
Might be worth making a `clamp = ( val, min, max ) => Math.min( Math.max( val, min ), max )` to reuse across all of these.

Reply:
Good idea. The whole codebase needs a good tidy up really!